OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,338 @@
+                }
+                thisCombination[i] = (int)code;
+                i++;
+            }
+            Array.Sort(thisCombination);
+            long hash = 0;
+            for (i = 0; i < MaxComboLength; i++)
+            {
+                hash = hash * MaxUsefulKey + thisCombination[i];
+            }
+            if (hash < MaxUsefulKey && _gameBindingCounter[hash] > 0)
+            {
+                return (int)RegistrationCode.CombinationTaken;
+            }
+            return (_comboRegistry.ContainsKey(hash) && !forRemoval ? (int)RegistrationCode.CombinationTaken : hash);
+        }
+
+        private RegistrationCode SwapCombination(ModInputCombination combination, bool toUnregister)
+        {
+            if (combination == null || combination.Combo == null)
+            {
+                _logger.Log("combination is null");
+                return RegistrationCode.InvalidCombination;
+            }
+            if (!_comboRegistry.ContainsValue(combination) && toUnregister)
+            {
+                return RegistrationCode.InvalidCombination;
+            }
+            var hashes = new List<long>();
+            foreach (var combo in combination.Combo.ToLower().Split('/'))
+            {
+                var hash = ParseCombination(combo, toUnregister);
+                if (hash <= 0)
+                {
+                    return (RegistrationCode)hash;
+                }
+                hashes.Add(hash);
+            }
+            combination.ClearSingles();
+            foreach (long hash in hashes)
+            {
+                if (toUnregister)
+                {
+                    _comboRegistry.Remove(hash);
+                    continue;
+                }
+                _comboRegistry.Add(hash, combination);
+                if (hash < MaxUsefulKey)
+                {
+                    combination.AddSingle((KeyCode)hash);
+                }
+            }
+            _logger.Log("succesfully registered " + combination.Combo);
+            return RegistrationCode.AllNormal;
+        }
+
+        public RegistrationCode RegisterCombination(IModInputCombination combination)
+        {
+            return SwapCombination((ModInputCombination)combination, false);
+        }
+
+        public RegistrationCode UnregisterCombination(IModInputCombination combination)
+        {
+            return SwapCombination((ModInputCombination)combination, true);
+        }
+
+        internal void SwapGamesBinding(InputCommand binding, bool toUnregister)
+        {
+            if (_gameBindingRegistry.Contains(binding) ^ toUnregister)
+            {
+                return;
+            }
+            var fields = binding is SingleAxisCommand ?
+                typeof(SingleAxisCommand).GetFields(NonPublic) : typeof(DoubleAxisCommand).GetFields(NonPublic);
+            foreach (var field in fields)
+            {
+                if (field.FieldType == typeof(KeyCode))
```

Comment:
You could do this, to combine loop+if:
```
var keyCodeFields = fields.Where(field => field.FieldType == typeof(KeyCode));
foreach(var keyCodeField in keyCodeFields)
```

Reply:
what do I need to add to usings to do that?

Reply:
nvm, unfortunately it seems to be only doable with List, and conversion to List seems to be non-trivial and IMO would make it harder to read

Reply:
It's fine, just a suggestion for shortening it. Nevermind :)